packages/node/src/login/oidc/ClientRegistrar.spec.ts

```diff
@@ -41,23 +48,239 @@ describe("ClientRegistrar", () => {
+        },
+      };
+      Issuer.discover = jest.fn().mockResolvedValueOnce(mockedIssuer);
+
+      // Run the test
+      const clientRegistrar = getClientRegistrar();
+      const client = await clientRegistrar.getClient(
+        {
+          sessionId: "mySession",
+          redirectUrl: "https://example.com",
+        },
+        {
+          ...IssuerConfigFetcherFetchConfigResponse,
+        }
+      );
+      expect(client.clientId).toEqual(mockDefaultClientConfig().client_id);
+      expect(client.clientSecret).toEqual(
+        mockDefaultClientConfig().client_secret
+      );
+    });
+
+    it("retrieves a registration bearer token if present in storage", async () => {
+      // Sets up the mock-up for DCR
+      const { Issuer } = jest.requireMock("openid-client");
+      const mockedIssuer = {
+        metadata: mockDefaultIssuerConfig(),
+        Client: {
+          register: jest.fn().mockResolvedValueOnce({
+            metadata: mockDefaultClientConfig(),
+          }),
+        },
+      };
+      Issuer.discover = jest.fn().mockResolvedValueOnce(mockedIssuer);
+      const mockStorage = mockStorageUtility(
+        {
+          mySession: {
+            registrationAccessToken: "a token",
+          },
+        },
+        false
+      );
+
+      // Run the test
+      const clientRegistrar = getClientRegistrar({
+        storage: mockStorage,
+      });
+      await clientRegistrar.getClient(
+        {
+          sessionId: "mySession",
+          redirectUrl: "https://example.com",
+        },
+        {
+          ...IssuerConfigFetcherFetchConfigResponse,
+        }
+      );
+      expect(mockedIssuer.Client.register).toHaveBeenCalledWith(
+        {
+          redirect_uris: ["https://example.com"],
+        },
+        {
+          initialAccessToken: "a token",
+        }
+      );
+    });
+
+    it("uses a registration bearer token if provided", async () => {
+      // Sets up the mock-up for DCR
+      const { Issuer } = jest.requireMock("openid-client");
+      const mockedIssuer = {
+        metadata: mockDefaultIssuerConfig(),
+        Client: {
+          register: jest.fn().mockResolvedValueOnce({
+            metadata: mockDefaultClientConfig(),
+          }),
+        },
+      };
+      Issuer.discover = jest.fn().mockResolvedValueOnce(mockedIssuer);
+
+      // Run the test
+      const clientRegistrar = getClientRegistrar();
+      await clientRegistrar.getClient(
+        {
+          sessionId: "mySession",
+          redirectUrl: "https://example.com",
+          registrationAccessToken: "a token",
+        },
+        {
+          ...IssuerConfigFetcherFetchConfigResponse,
+        }
+      );
+      expect(mockedIssuer.Client.register).toHaveBeenCalledWith(
+        {
+          redirect_uris: ["https://example.com"],
+        },
+        {
+          initialAccessToken: "a token",
+        }
+      );
+    });
+
+    it("saves the registered client information in storage", async () => {
```

Comment:
I'm not sure, but this test setup and execution seems identical to previous tests, with just the final two assertions being different. If that is the case, I wouldn't duplicate so much code - I'd just move these two assertions up into a previous test. So yes, it would be a single test testing for multiple behaviours, but I'd be fine with that (over code duplication), and if we really wanted to split it out into multiple independent tests (although I don't see a justification yet), then I'd refactor the test setup and execution into it's own callable static function.